searchlib/src/vespa/searchlib/attribute/enumstore.hpp

```diff
@@ -65,14 +65,18 @@ EnumStoreT<EntryType>::load_unique_value(const void* src, size_t available, Inde
     idx = _store.get_allocator().allocate(*value);
 
     if (prev_idx.valid()) {
-        assert(ComparatorType::compare(getValue(prev_idx), *value) < 0);
+        auto cmp = make_comparator(*value);
+        assert(cmp(prev_idx, Index()));
     }
     return sizeof(DataType);
 }
 
 template <typename EntryType>
 EnumStoreT<EntryType>::EnumStoreT(bool has_postings)
-    : _store(make_enum_store_dictionary(*this, has_postings, EntryType::hasFold() ? std::make_unique<FoldedComparatorType>(*this) : std::unique_ptr<datastore::EntryComparator>())),
+    : _store(make_enum_store_dictionary(*this, has_postings,
+                                        (has_string_type() ?
+                                         std::make_unique<FoldedComparatorType>(_store.get_data_store()) :
```

Comment:
This does not compile when using clang, due to _store not being initialized:
`
vespa/searchlib/attribute/enumstore.hpp:78:81: error: field '_store' is uninitialized when used here [-Werror,-Wuninitialized]
`

Reply:
Fixed